metrics-collectd/src/test/java/com/codahale/metrics/collectd/Receiver.java

```diff
@@ -0,0 +1,81 @@
+package com.codahale.metrics.collectd;
+
+import org.collectd.api.Notification;
+import org.collectd.api.ValueList;
+import org.collectd.protocol.Dispatcher;
+import org.collectd.protocol.UdpReceiver;
+import org.junit.rules.ExternalResource;
+
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+public final class Receiver extends ExternalResource implements Runnable {
+
+    private int port;
+    private UdpReceiver receiver;
+    private Thread receiverThread;
+    private BlockingQueue<ValueList> queue;
+
+    public Receiver(int port) {
+        this.port = port;
+    }
+
+    @Override
+    protected void before() throws Throwable {
+        queue = new LinkedBlockingQueue<ValueList>();
+        receiver = new UdpReceiver(new QueueDispatcher(queue));
+        receiver.setPort(port);
+        start();
+    }
+
+    private void start() {
+        receiverThread = new Thread(this);
+        receiverThread.start();
+    }
+
+    @Override
+    public void run() {
+        try {
+            receiver.listen();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    public ValueList next() throws InterruptedException {
+        return queue.poll(1, TimeUnit.SECONDS);
+    }
+
+    @Override
+    protected void after() {
+        stop();
+    }
+
+    private void stop() {
+        receiver.shutdown();
+        try {
+            receiverThread.join();
```

Comment:
It doesn't seem useful to `join` with the receiver thread. The main thread doesn't expect any results from it. We can just call an `interrupt` to signal the receiver thread to exit if it's stuck listening on the socket.